pkg/controller/sync/aggregator/aggregator.go

```diff
@@ -0,0 +1,133 @@
+package aggregator
+
+import (
+	"fmt"
+	"maps"
+	gosync "sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	syncset    = "TODOa"
+	configsync = "TODOb"
+)
+
+type GVKAgreggator interface {
+	UpsertWithValidation(k Key, gvks []schema.GroupVersionKind) (bool, error)
+	Remove(k Key) error
+	GetGVKs(k Key) ([]schema.GroupVersionKind, bool)
+	ListKeys() []Key
+	ListGVKs() []schema.GroupVersionKind
+}
+
+type Key struct {
+	Kind string
+	Name string
+}
+
+func (k *Key) ValidateKind() error {
+	if k.Kind != syncset && k.Kind != configsync {
+		return fmt.Errorf("unsupported key kind for (%s, %s); supported values are: %s", k.Kind, k.Name, "a, b")
+	}
+
+	return nil
+}
+
+func NewGVKAggregator() GVKAgreggator {
+	return &bidiGVKAggregator{
+		store:        make(map[Key]map[schema.GroupVersionKind]struct{}),
+		reverseStore: make(map[schema.GroupVersionKind]map[Key]struct{}),
+	}
+}
+
+// todo comments
+type bidiGVKAggregator struct {
+	mu gosync.RWMutex
+
+	store        map[Key]map[schema.GroupVersionKind]struct{}
+	reverseStore map[schema.GroupVersionKind]map[Key]struct{}
+}
+
+// GetGVKs implements GVKAgreggator
```

Comment:
deduping this and the `ListKeys` question below

I can take out these funcs since they won't be used by the controller or watch manager but I do think they may end up being used by `gator` things for verbosity and shift left explanations of why a `requireSyncData` annotation is not satisfied. I think we can add those funcs back in if we end up needing them in that case.